runtime/reconcilers/alert.go

```diff
@@ -724,23 +737,33 @@ func (r *AlertReconciler) popCurrentExecution(ctx context.Context, self *runtime
 	var notificationErr error
 	var sentNotifications bool
 	if msg != nil {
-		if adminMeta != nil {
-			// Note: adminMeta may not always be available (if outside of cloud). In those cases, we leave the links blank (no clickthrough available).
-			openLink, err := addExecutionTime(adminMeta.OpenURL, executionTime)
-			if err != nil {
-				return fmt.Errorf("failed to build open url: %w", err)
-			}
-			msg.OpenLink = openLink
-			msg.EditLink = adminMeta.EditURL
-		}
-
 		for _, notifier := range a.Spec.Notifiers {
 			switch notifier.Connector {
 			// TODO: transform email client to notifier
 			case "email":
 				recipients := pbutil.ToSliceString(notifier.Properties.AsMap()["recipients"])
 				for _, recipient := range recipients {
 					msg.ToEmail = recipient
+
+					// Set recipient-specific URLs if available from admin metadata
+					if adminMeta != nil && adminMeta.RecipientURLs != nil {
+						if recipientURLs, ok := adminMeta.RecipientURLs[recipient]; ok {
+							// Use recipient-specific URLs (with magic token)
+							openLink, err := addExecutionTime(recipientURLs.OpenURL, executionTime)
+							if err != nil {
+								return fmt.Errorf("failed to build recipient open url: %w", err)
+							}
+							msg.OpenLink = openLink
+							msg.EditLink = recipientURLs.EditURL
+							msg.UnsubscribeLink = recipientURLs.UnsubscribeURL
+						} else {
+							// Fallback to default URLs (for internal users)
+							msg.OpenLink = ""
+							msg.EditLink = ""
+							msg.UnsubscribeLink = ""
```

Comment:
"for internal users" – what does that mean? Before there was a comment about outside of cloud:
```
// Note: adminMeta may not always be available (if outside of cloud). In those cases, we leave the links blank (no clickthrough available).
```